tests/test_server.py

```diff
@@ -189,11 +206,17 @@ async def test_metric_names(mcp: OptunaMCP, metric_names: list[str]) -> None:
     await mcp.call_tool("set_metric_names", arguments={"metric_names": metric_names})
     result = await mcp.call_tool("get_metric_names", arguments={})
     assert isinstance(result, Sequence)
-    assert len(result) == 1
-    assert isinstance(result[0], TextContent)
-    r = json.loads(":".join(result[0].text.split(":")[1:]).strip())
-    assert isinstance(r, list)
-    assert r == metric_names
+    assert len(result) == 2
+    assert isinstance(result[0], list)
+    assert isinstance(result[0][0], TextContent)
+    assert isinstance(result[1], dict)
+    user_attrs_from_text = result[0][0].text
+    user_attrs_from_dict = result[1]["result"]
+
+    for user_attrs in (user_attrs_from_text, user_attrs_from_dict):
+        r = json.loads(":".join(user_attrs.split(":")[1:]).strip())
+        assert isinstance(r, list)
+        assert r == metric_names
```

Comment:
Hmm, I felt the output data was not structured when I saw these lines for the result parsing. Not a strong opinion, but we might include the metric names in `StudyResponse` as I mentioned above.